src/main/java/com/gradle/GitMetadata.java

```diff
@@ -0,0 +1,42 @@
+package com.gradle;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static com.gradle.Utils.execAndGetStdOut;
+
+class GitMetadata {
+
+    private String[] args;
+    boolean isGitInstalled;
+    private Supplier<Optional<String>> fromEnv;
+
+    private GitMetadata(boolean isGitInstalled) {
+        this.isGitInstalled = isGitInstalled;
+    }
+
+    static GitMetadata given(boolean isGitInstalled) {
```

Comment:
i'm not a fan of the name `given` here for the static constructor method since it isn't very descriptive when used inline.